torch/csrc/utils/tensor_flatten.cpp

```diff
@@ -0,0 +1,95 @@
+#include "torch/csrc/utils/tensor_flatten.h"
+
+#include <unordered_map>
+
+namespace torch { namespace utils {
+
+using namespace at;
+
+std::vector<TensorGroup> take_tensors(const TensorList& tensors, std::size_t size_limit) {
+  std::vector<TensorGroup> results;
+  results.reserve(tensors.size()); // an overapproximation, but at least we won't have to copy stuff around
+  std::unordered_map<at::Type*, TensorGroup> groups_;
```

Comment:
Not important, but why the `_` suffix?

Reply:
Oh I implemented a generator-like thing (basically a lazy iterator), and this was part of the state that it was closed over. It wasn't any faster, and the memory complexity is the same, so I just used the simpler version that precomputes the whole vector at once. I'll remove the underscore